src/shims/unix/freebsd/foreign_items.rs

```diff
@@ -26,17 +26,29 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                     this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
                 this.write_null(dest)?;
             }
-            "pthread_set_name_np" => {
+            // Old vs new name
+            "pthread_set_name_np" | "pthread_setname_np" => {
                 let [thread, name] =
                     this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
-                let max_len = usize::MAX; // freebsd does not seem to have a limit.
+                let max_len = 20; // freebsd silently truncates to MAXCOMLEN (+ 1).
```

Comment:
And MAXCOMLEN is 19 then, I assume?

Reply:
Yes

Reply:
Is there somewhere you can link to that documents this?

Reply:
Sure
https://github.com/freebsd/freebsd-src/blob/314542de6d0a044640d349d68b2d1a1fe5f427ff/sys/sys/param.h#L127

here the syscall doc

https://github.com/freebsd/freebsd-src/blob/314542de6d0a044640d349d68b2d1a1fe5f427ff/lib/libc/sys/thr_set_name.2#L47

Reply:
Okay, please add those links to the code. :)